application_sandbox/async_compute/main.cpp

```diff
@@ -419,35 +464,54 @@ class ASyncThreadRunner {
     }
 
     std::lock_guard<std::mutex> lock(data_mutex_);
-    if (mailbox_buffer_ == -1) {
+    if (mailbox_buffers_.empty()) {
       // Nothing is ready;
       return index;
     }
 
-    int32_t mb = mailbox_buffer_;
-    mailbox_buffer_ = -1;
+    int32_t mb = mailbox_buffers_.back();
+    mailbox_buffers_.pop_back();
     if (index != -1) {
-      // Enqueues a command-buffer that transitions the buffer back to
-      // the compute queue. It also sets the fence that we can wait on
-      // in the future.
-      auto& data = data_[index];
+      mailbox_buffers_.push_back(index);
+    }
+    std::vector<VkCommandBuffer> command_buffers;
```

Comment:
Can't this be move to (and replace) line 480?

Reply:
Do you mean move the declaration of the vector inside the loop? That would cause the memory allocated for the vector's contents to be freed and re-allocated every loop. My understanding is that this way (declared outside, `clear()` on each loop) will re-use the memory (just setting the size to 0).

Reply:
Right, from a performance point of view that is correct. I was thinking more for a readability point of view. It's fine the way it is.